test/Interop/Cxx/class/Inputs/member-variables.h

```diff
@@ -24,6 +27,14 @@ struct HasZeroSizedField {
   void set_c(short c) { this->c = c; }
 };
 
+struct ReuseFieldPadding {
+  [[no_unique_address]] std::optional<int> a;
```

Comment:
Can you add some getter/setter methods to ensure this field is still accessible via those?

Reply:
Also, can you add another test that doesn't really on the layout of `std::optional`, i.e., using your own padded type?

Reply:
> Can you add some getter/setter methods to ensure this field is still accessible via those?

Added a getter.

> Also, can you add another test that doesn't really on the layout of std::optional, i.e., using your own padded type?

I tried and failed. Not sure what is unique about `std::optional` but my couple of attempts to make the compiler do the same optimisation with my own type did not work. Will need to look deeper to figure out what is the difference because I tried to recreate the same layout.